src/App.tsx

```
@@ -1,27 +1,51 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.scss';
+
 import { GoodsList } from './GoodsList';
 
 // import { getAll, get5First, getRed } from './api/goods';
 // or
-// import * as goodsAPI from './api/goods';
+import * as goodsAPI from './api/goods';
+
+import { Good } from './types/Good';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+
+  const loadAllGoods = () => {
+    goodsAPI.getAll().then(setGoods);
+  };
+
+  const load5FirstGoods = () => {
+    goodsAPI.get5First().then(setGoods);
+  };
+
+  const loadRedGoods = () => {
+    goodsAPI.getRedGoods().then(setGoods);
+  };
```

All three handlers call the API and only do `.then(setGoods)` without `.catch` or try/catch. That means network/HTTP errors are not handled and the UI won't reflect failures. This violates checklist item #3: "When you make a request to the server, don't forget to handle possible errors". Also, the handlers are recreated on every render (not memoized) which violates checklist item #1: "Make sure that you add some rendering optimization techniques (`React.memo` and so on.)". Please:

- Handle errors in each handler (or ensure the API methods reject on non-OK responses and catch here). Use `try { setLoading(true); const data = await goodsAPI.getAll(); setGoods(data); } catch (e) { setError(String(e)); } finally { setLoading(false); }` or `.then(...).catch(...).finally(...)`.
- Stabilize the handlers using `useCallback` so they have stable identities, e.g. `const loadAllGoods = useCallback(async () => { ... }, []);` to avoid unnecessary child re-renders. 

Current handlers are defined at these lines in src/App.tsx .